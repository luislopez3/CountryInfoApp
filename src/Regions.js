import React from "react";
import { Link, Route, useRouteMatch } from "react-router-dom";
import Countries from "./Countries";

export default function Regions() {
  const regionList = ["Africa", "Americas", "Asia", "Europe", "Oceania"];
  const { url, path } = useRouteMatch();

  return (
    <div>
      <h1>Regions</h1>
      <p>Select a region to view the Countries within it.</p>
      {regionList.map((region, index) => {
        return (
          <div className="country-row">
            <ul className="country-column" key={index}>
              <Link to={`${url}/${region}`}>
                <button>{region}</button>
              </Link>
            </ul>
          </div>
        );
      })}
      <hr />
      <Route path={`${path}/:region`}>
        <Countries />
      </Route>
    </div>
  );
}
