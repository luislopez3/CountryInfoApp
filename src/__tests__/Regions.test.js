import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Regions from '../Regions.js';

describe("Show Details component", () => {
    test("Matches the snapshot", () => {
        const wrapper = create(
        <MemoryRouter ><Regions /></MemoryRouter>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});