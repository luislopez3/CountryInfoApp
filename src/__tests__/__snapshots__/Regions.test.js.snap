// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Details component Matches the snapshot 1`] = `
<>
  <h1>
    Regions
  </h1>
  <p>
    Select a region to view the Countries within it.
  </p>
  <ul>
    <a
      href="//Africa"
      onClick={[Function]}
    >
      <button>
        Africa
      </button>
    </a>
  </ul>
  <ul>
    <a
      href="//Americas"
      onClick={[Function]}
    >
      <button>
        Americas
      </button>
    </a>
  </ul>
  <ul>
    <a
      href="//Asia"
      onClick={[Function]}
    >
      <button>
        Asia
      </button>
    </a>
  </ul>
  <ul>
    <a
      href="//Europe"
      onClick={[Function]}
    >
      <button>
        Europe
      </button>
    </a>
  </ul>
  <ul>
    <a
      href="//Oceania"
      onClick={[Function]}
    >
      <button>
        Oceania
      </button>
    </a>
  </ul>
  <hr />
</>
`;
